app/services/assessmentcentre/AssessmentCentreService.scala

```diff
@@ -19,15 +19,21 @@ package services.assessmentcentre
 import common.FutureEx
 import model.{ ProgressStatuses, SerialUpdateResult }
 import model.command.ApplicationForFsac
+import model.persisted.fsac.{ AnalysisExercise, AssessmentCentreTests }
+import play.api.Logger
 import repositories.application.GeneralApplicationRepository
 import repositories.assessmentcentre.{ AssessmentCentreMongoRepository, AssessmentCentreRepository }
+import services.assessmentcentre.AssessmentCentreService.CandidateAlreadyHasAnAnalysisExerciseException
 
 import scala.concurrent.Future
 import scala.concurrent.ExecutionContext.Implicits.global
 
 object AssessmentCentreService extends AssessmentCentreService {
   val applicationRepo = repositories.applicationRepository
   val assessmentCentreRepo = repositories.assessmentCentreRepository
+
+  case class CandidateAlreadyHasAnAnalysisExerciseException(message: String) extends Exception(message)
+  case class CandidateAlreadyHasNoAnalysisExerciseException(message: String) extends Exception(message)
```

Comment:
Is this named correctly (e.g. should it be "CandidateHasNoAnalysisExerciseException") ?